platform/platform_tests/duration_tests.cpp

```diff
@@ -0,0 +1,106 @@
+#include "testing/testing.hpp"
+
+#include "platform/duration.hpp"
+
+#include <chrono>
+
+namespace platform
+{
+void AddUnit(long value, Duration::Units unit, std::string & result)
+{
+  if (value == 0)
+    return;
+  if (!result.empty())
+    result += kNonBreakingSpace;
+  result += std::to_string(value);
+  result += kNarrowNonBreakingSpace;
+  result += Duration::GetUnitsString(unit);
+}
+
+struct TestData
+{
+  std::chrono::days m_days;
+  std::chrono::hours m_hours;
+  std::chrono::minutes m_minutes;
+
+  TestData(long days, long hours, long minutes)
+    : m_days(days), m_hours(hours), m_minutes(minutes)
+  {}
+
+  long Seconds() const
+  {
+    return (std::chrono::duration_cast<std::chrono::seconds>(m_days) +
+            std::chrono::duration_cast<std::chrono::seconds>(m_hours) +
+            std::chrono::duration_cast<std::chrono::seconds>(m_minutes)).count();
+  }
+
+  std::string ToString()
```

Comment:
We use external `std::string DebugPrint(TestData const & data)` function (outside of the class), then you don't need to explicitly call `ToString()` from TEST_* and ASSERT_* macro.

Reply:
In this case, this is almost a copy-paste code from the implementation. And if there's a bug in both places, then test will always succeed.

Would it be clearer to explicitly store expected results strings in TestData struct?